test.py

```diff
@@ -427,6 +427,8 @@ async def reap(done, pending, signaled):
             result = run_pytest(options, run_id=i)
             total_tests += result[0]
             failed_tests.extend(result[1])
+            if len(failed_tests) >= max_failures != 0:
+                break
```

Comment:
In other places we print `print("Too much failures, stopping")`, shouldn't we print it here as well?

Reply:
Maybe we should consider adjusting the `maxfail` parameter passed to pytest if there were some failures in the previous runs of pytest? For example, if we have 10 allowed failures in the beginning and in the first repeat we get 6 failures, then the second run should be run with `maxfail=4`, right?

Reply:
> I doubt that is needed. This is a temporary solution before providing the native repeats. So having this is fine for me. 

If we are going to have a pytest-native solution in the end then I'm fine with this, too.